client/src/visualiser-src/sorting-visualiser/data-structure/GraphicalSorts.ts

```diff
@@ -104,7 +109,132 @@ export default class GraphicalSortList extends GraphicalDataStructure {
     return producer;
   }
 
-  insertion(): AnimationProducer {
+  public merge(): AnimationProducer {
+    const producer = new SortsMergeAnimationProducer();
+
+    producer.renderMergeCode();
+
+    const low = 0;
+    const high = this.elementList.length - 1;
+
+    const p = 0;
+
+    const tmpList = [...this.elementList];
+
+    this.mergeSort(producer, low, high, low, tmpList);
+
+    return producer;
+  }
+
+  public mergeSort(producer: SortsMergeAnimationProducer, low: number, high: number, position: number, tmpList: GraphicalSortsElement[]) {
+    if (high <= low) {
+      return;
+    }
+
+    const mid = Math.floor((low + high) / 2);
+
+    let p = position;
+
+    this.mergeSort(producer, low, mid, low, tmpList);
+    this.mergeSort(producer, mid+1, high, mid+1, tmpList);
+
+    let pointerLeft = low;
+    let pointerRight = mid + 1;
+
+    producer.doAnimationAndHighlightTimestamp(
+      21,
+      false,
+      producer.highlightSorting,
+      low,
+      high,
+      this.elementList
+    );
+    
+    // compare the value on the pointerLeft and value on the pointerRight
+    // move down the column with smaller value
+    // and increment the index we are checking by 1
+    // do not change the index of column with larger value
+    // if either left or right is finished, then move all the other parts down
+    // finally move all the columns up
+
+    while (pointerLeft <= mid && pointerRight <= high) {
+      // compare this.elemensList[pointerLeft] and this.elementList[pointerRight]
```

Comment:
These comments are also unnecessary because you can tell what's happening from the code you've written.